plugins/parsers/json/parser.go

```diff
@@ -54,10 +59,28 @@ func (p *JSONParser) parseObject(metrics []telegraf.Metric, jsonOut map[string]i
 		return nil, err
 	}
 
-	tags, nFields := p.switchFieldToTag(tags, f.Fields)
+	//if time key is specified, set it to time
+	nTime := time.Now().UTC()
+	if p.JSONTimeKey != "" {
+		if p.JSONTimeFormat == "" {
+			err := fmt.Errorf("E! If 'json_time_key' is specified, there must be a 'json_time_format'")
```

Comment:
Since this is an error and not a log message, don't include the `E!`.  As a nitpick, it would probably be good to try to come up with a shorter error string, perhaps: `use of 'json_time_key' requires 'json_time_format'`?

Remember to do this for the other errors, here is some more info on how "standard" Go error strings should look: https://github.com/golang/go/wiki/CodeReviewComments#error-strings